test/00-setup.js

```diff
@@ -2,31 +2,43 @@
 const tape = require('tape')
 const promisifyTape = require('tape-promise').default
 const test = promisifyTape(tape)
-const makeDir = require('make-dir')
-const fs = require('fs')
+const _ = require('lodash')
+const MemoryFS = require('memory-fs')
+const _fs = new MemoryFS()
+const fs = _.mapValues(Object.getPrototypeOf(_fs), (v, k) => typeof v === 'function' ? v.bind(_fs) : v)
 const path = require('path')
-const fixtures = path.resolve(__dirname, 'fixtures')
 
-const srcs = {
-  'src.exe': 'exe',
-  'src.env': '#!/usr/bin/env node\nconsole.log(/hi/)\n',
-  'src.env.args': '#!/usr/bin/env node --expose_gc\ngc()\n',
-  'src.sh': '#!/usr/bin/sh\necho hi\n',
-  'src.sh.args': '#!/usr/bin/sh -x\necho hi\n'
+const {fixtures, fixtures2} = process.platform === 'win32' ? {
```

Comment:
`is-windows` is in the deps, so you may use that

Reply:
Should I use `is-windows` in this case?  The conditional here is equivalent to testing if `path.win32 === path`, which will be true on Windows but not on Cygwin, whereas I think `is-windows` returns true for Cygwin.

What I'm really trying to test here is not if we're on Windows but if we're running a Windows build of nodejs, since that affects which `path` implementation is used by default.

On Linux, `path.relative()` can always return a relative path, even if it means `..`ing to the root of the filesystem.  On Windows, `path.relative()` can't do that when the drive is different, so we're able to call `path.relative()` and then check if the result `isAbsolute()`.

Does cmd-shim need to create .cmd shims from within a Linux build of nodejs?  (E.g. on WSL or Cygwin)  If so, I'll need to tweak the code to special-case this.  To properly support WSL, we'll need to call the external `wslpath` binary to convert the paths to Windows, then use `path.win32` to manipulate them.

Reply:
I don't know the answer to this. I think the path.relative tweaks can be done in a separate PR. As far as I understand, this PR does not change how path.relative is used

Reply:
Without those tweaks, this PR won't be doing its job of allowing cross-drive linking.  Here's an example:

If I want to link from C:\foo to D:\bar, and I run it on Windows, `path.relative('C:\\', 'D:\\bar')` will return 'D:\bar'.  `path.isAbsolute()` will return true, and the %~dp0 prefix will be omitted from the path in the shim.

If I do the same in Cygwin or WSL, `path.relative('/c', '/d/bar')` will return '../d/bar'.  `path.isAbsolute()` will return false, and the `%~dp0` prefix will be included in the shim, so the shim will fail on Windows.  `C:\..\d\bar` is not what we want.

I think the question is: do we want to support creating cross-drive links from a Linux build of node running in WSL and/or Cygwin?  If "no", then this PR does the trick already.  If "yes", we have some more work to do.

Reply:
Currently, it doesn't work on Windows at all, so this is already a huge improvement, as it is